File: namsor/csharp/src/OSEG.NamsorExamples/GenderJapaneseNamePinyinExample.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text.Json;

using App.Namsor.Api;
using App.Namsor.Client;
using App.Namsor.Model;

namespace OSEG.NamsorExamples;

public class GenderJapaneseNamePinyinExample
{
    public static void Run()
    {
        var config = new Configuration();
        config.ApiKey.Add("X-API-KEY", "YOUR_API_KEY");

        try
        {
            var response = new JapaneseApi(config).GenderJapaneseNamePinyin(
                japaneseSurname: "中松",
                japaneseGivenName: "義郎"
            );

            Console.WriteLine(response);
        }
        catch (ApiException e)
        {
            Console.WriteLine("Exception when calling JapaneseApi#GenderJapaneseNamePinyin: " + e.Message);
            Console.WriteLine("Status Code: " + e.ErrorCode);
            Console.WriteLine(e.StackTrace);
        }
    }
}

File: namsor/typescript-axios/src/GenderJapaneseNamePinyinExample.ts
import axios, { AxiosError } from "axios";
import * as api from "namsor_client"

const configuration = new api.Configuration({
  apiKey: "YOUR_API_KEY",
});

new api.JapaneseApi(configuration).genderJapaneseNamePinyin(
  "中松", // japaneseSurname
  "義郎", // japaneseGivenName
).then(response => {
  console.log(response.data);
}).catch((error: Error | AxiosError) => {
  console.log("Exception when calling JapaneseApi#genderJapaneseNamePinyin:");
  axios.isAxiosError(error) ? console.log(error.message) : console.log(error);
});
